openapi-cli/src/test/java/io/ballerina/openapi/cmd/OpenAPICmdTest.java

```diff
@@ -80,68 +80,188 @@ public void testSuccessfulServiceGeneration() throws IOException {
         new CommandLine(cmd).parseArgs(args);
         cmd.execute();
         Path expectedSchemaFile = resourceDir.resolve(Paths.get("expected_gen", "petstore_schema.bal"));
-
         Stream<String> expectedSchemaLines = Files.lines(expectedSchemaFile);
         String expectedSchemaContent = expectedSchemaLines.collect(Collectors.joining("\n"));
         expectedSchemaLines.close();
         if (Files.exists(this.tmpDir.resolve("client.bal")) &&
                 Files.exists(this.tmpDir.resolve("petstore_service.bal")) &&
-                Files.exists(this.tmpDir.resolve("types.bal"))) {
+                Files.exists(this.tmpDir.resolve("types.bal")) &&
+                Files.exists(this.tmpDir.resolve("tests/test.bal"))) {
             //Compare schema contents
             Stream<String> schemaLines = Files.lines(this.tmpDir.resolve("types.bal"));
             String generatedSchema = schemaLines.collect(Collectors.joining("\n"));
             schemaLines.close();
+            generatedSchema = (generatedSchema.trim()).replaceAll("\\s+", "");
+            expectedSchemaContent = (expectedSchemaContent.trim()).replaceAll("\\s+", "");
+            if (expectedSchemaContent.equals(generatedSchema)) {
+                Assert.assertTrue(true);
+                deleteGeneratedFiles(false);
+            } else {
+                Assert.fail("Expected content and actual generated content is mismatched for: "
+                        + petstoreYaml.toString());
+                deleteGeneratedFiles(false);
+            }
+        } else {
+            Assert.fail("Service generation failed.");
+        }
+    }
 
+    @Test(description = "Test openapi to ballerina generation with license headers")
+    public void testGenerationWithLicenseHeaders() throws IOException {
+        Path petstoreYaml = resourceDir.resolve(Paths.get("petstore.yaml"));
+        Path licenseHeader = resourceDir.resolve(Paths.get("license.txt"));
+        String[] args = {"--input", petstoreYaml.toString(), "-o", this.tmpDir.toString(), "--prefix",
+                licenseHeader.toString()};
+        OpenApiCmd cmd = new OpenApiCmd(printStream, tmpDir, false);
+        new CommandLine(cmd).parseArgs(args);
+        cmd.execute();
+        Path expectedSchemaFile = resourceDir.resolve(Paths.get("expected_gen",
+                "petstore_schema_with_license.bal"));
+        Stream<String> expectedSchemaLines = Files.lines(expectedSchemaFile);
+        String expectedSchemaContent = expectedSchemaLines.collect(Collectors.joining("\n"));
+        expectedSchemaLines.close();
```

Comment:
Should we close this in a filnally block? If `String expectedSchemaContent = expectedSchemaLines.collect(Collectors.joining("\n"));` would expectedSchemaLines get closed?

Reply:
@lnash94 WDYT ?

Reply:
+1 to have it inside the Finally Block. yeah, it is a safer way to recovering resources and preventing resource leaks.

Reply:
Added a try with resources. Please review